src/component/form.js

```diff
@@ -1,25 +1,54 @@
 import { useDispatch } from 'react-redux/es/exports';
-import React, { useRef } from 'react';
+import React, { useState, useEffect } from 'react';
 import uniqid from 'uniqid';
-import { addBook } from '../redux/books/books';
+import { addBook, fetchBooks } from '../redux/books/books';
 import './books.css';
 
 const Form = () => {
   const dispatch = useDispatch();
-  const title = useRef(null);
-  const author = useRef(null);
+  const [state, setState] = useState({
+    item_id: uniqid(),
+    title: '',
+    author: '',
+    category: '',
+  });
+
+  const handleChanges = (e) => {
+    setState({
+      ...state, [e.target.name]: e.target.value,
+    });
+  };
+
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, []);
```

Comment:
- [ ] The `fetchBooks()` action you have dispatched here makes an async call. Can you help me understand why we need to fetch the books from the API when the child component is rendered? I would say it is enough to fetch the list of books from the API when the container component (books.js in your case) is rendered.